src/db/mysql/models.rs

```diff
@@ -514,6 +518,10 @@ impl MysqlDb {
             query = query.filter(bso::id.eq_any(ids));
         }
 
+        // it's possible for two BSOs to be inserted with the same `modified` date,
+        // since there's no guarantee of order when doing a get, pagination can return
+        // an error. We "fudge" a bit here by taking the id order as a secondary, since
+        // that is guaranteed to be unique by the client.
```

Comment:
Yikes, I finally see what you mean about the race condition this caused in the tests here.